scripts/clippy-all

```diff
@@ -20,24 +20,18 @@ ALLOWED_CLIPPY_LINTS=(-A clippy::module_inception  \
 	-A clippy::new_ret_no_self \
 	-A clippy::unnecessary_sort_by)
 
-if [[ "${TIKV_ENABLE_FEATURES}" == *prost-codec* ]] ; then
-	cargo clippy --workspace --all-targets --no-default-features \
-		--exclude cdc --exclude backup --exclude tests --exclude cmd \
-		--exclude fuzz-targets --exclude fuzzer-honggfuzz --exclude fuzzer-afl --exclude fuzzer-libfuzzer \
+cargo clippy --workspace --all-targets --no-default-features \
```

Comment:
The original style is on purpose to speed up the checking speed.

Reply:
But it doesn't cover all code snippets.

Reply:
Even after adding `--all-targets`?

Reply:
Tried to remove the for loop and found more warnings.😅